kvmagent/kvmagent/test/vmmeta_testsuite/local/test_file_metadata_handler.py

```
@@ -0,0 +1,430 @@
+            f.write('NOT JSON')
+
+        entries = self.handler._do_scan(self.tmpdir)
+        self.assertEqual(len(entries), 1)
+        self.assertEqual(entries[0].vmName, '')
+
+    def _test_scan_empty_directory(self):
+        entries = self.handler._do_scan(self.tmpdir)
+        self.assertEqual(len(entries), 0)
+
+    def _test_scan_missing_directory(self):
+        entries = self.handler._do_scan('/tmp/nonexistent_vmmeta_dir_xyz')
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/nonexistent_vmmeta_dir_xyz"

(S108)